molecule_docker/playbooks/network/delete_network.yml

```diff
@@ -0,0 +1,10 @@
+- name: Retrieve network info
+  docker_network_info:
+    name: "{{ item.name }}"
+  register: docker_netname
+
+- name: Delete docker network(s)
+  community.docker.docker_network:
+    name: "{{ item.name }}"
+    state: "absent"
+  when: docker_netname.exists and 'owner' in docker_netname.network.Labels and docker_netname.network.Labels.owner == 'molecule'
```

Comment:
split this into two lines as AND is implicit between them, make it easier to read the code.